deploy/deploy/operations.py

```diff
@@ -0,0 +1,92 @@
+import os
+import json
+import asyncio
+from typing import List,Dict,Any
+
+from dffml.df.base import op
+from .definitions import *
+from .exceptions import CannotRemoveContainer
+
+from dffml_feature_git.feature.operations import clone_git_repo
+from dffml_feature_git.util.proc import check_output, create, stop
+
+
+
+@op(
+    inputs = {"payload_str":git_payload_string},
+    outputs= {"payload":git_payload}
+)
+def get_payload(payload_str:str)->Dict[str,Any]:
+    payload = json.loads(payload_str)
+    return {"payload":payload}
+
+@op(
+    inputs={"payload":git_payload},
+    outputs={"url":clone_git_repo.op.inputs["URL"]}
+)
+def get_url_from_payload(payload:Dict[str,Any]):
+    return {
+        "url": payload["repository"]["clone_url"]
+    }
+
+@op(inputs={"payload":git_payload},outputs={"is_default_branch":is_default_branch})
+def check_if_default_branch(payload):
+    pushed_branch = payload["ref"].split("/")[-1]
+    default_branch = payload["repository"]["default_branch"]
+    return {
+        "is_deafault_branch":(default_branch==pushed_branch)
+    }
+
+@op(inputs={"payload":git_payload},outputs={"image_tag":docker_image_tag})
+def get_image_tag(payload):
+    url = payload["repository"]["html_url"] # eg:-"https://github.com/username/Hello-World",
+    tag = url.split("/")
+    tag = f"{tag[-2]}/{tag[-1]}"
+    return {"image_tag":tag}
+@op(
+    inputs={
+        "repo":clone_git_repo.op.outputs["repo"],
+        "image_tag":docker_image_tag
+        },
+    outputs={"build_status":is_image_built},
+    conditions=[check_if_default_branch]
```

Comment:
@pdxjohnny is the usage of conditions correct,or must the input to `condition_op` be same as that of `op` ?

Reply:
You want to use the definition

```suggestion
    conditions=[is_default_branch]
```